src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1277 @@
+                {
+                    var _ = swarm.ConnectAsync(remoteAddress).Result;
+                });
+            }
+            finally
+            {
+                swarm.StopAsync().Wait();
+            }
+        }
+
+        [TestMethod]
+        public void Connect_Failure_Event()
+        {
+            const string remoteId = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb";
+            MultiAddress remoteAddress = $"/ip4/127.0.0.1/tcp/4040/ipfs/{remoteId}";
+            var swarm = new SwarmService {LocalPeer = _self};
+            Peer unreachable = null;
+            swarm.PeerNotReachable += (s, e) => { unreachable = e; };
+            swarm.StartAsync().Wait();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() =>
+                {
+                    var _ = swarm.ConnectAsync(remoteAddress).Result;
+                });
+            }
+            finally
+            {
+                swarm.StopAsync().Wait();
+            }
+
+            Assert.IsNotNull(unreachable);
+            Assert.AreEqual(remoteId, unreachable.Id.ToBase58());
+        }
+
+        [TestMethod]
+        public void Connect_Not_Peer()
+        {
+            const string remoteId = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb";
+            MultiAddress remoteAddress = $"/dns/npmjs.com/tcp/80/ipfs/{remoteId}";
+            var swarm = new SwarmService {LocalPeer = _self};
+            swarm.StartAsync().Wait();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() =>
+                {
+                    var _ = swarm.ConnectAsync(remoteAddress).Result;
+                });
+            }
+            finally
+            {
+                swarm.StopAsync().Wait();
+            }
+        }
+
+        [TestMethod]
+        public void Connect_Cancelled()
+        {
+            var cs = new CancellationTokenSource();
+            cs.Cancel();
+            const string remoteId = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb";
+            MultiAddress remoteAddress = $"/ip4/127.0.0.1/tcp/4002/ipfs/{remoteId}";
+            var swarm = new SwarmService {LocalPeer = _self};
+            swarm.StartAsync().Wait(cs.Token);
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() =>
+                {
+                    var _ = swarm.ConnectAsync(remoteAddress, cs.Token).Result;
+                });
+            }
+            finally
+            {
+                swarm.StopAsync().Wait(cs.Token);
+            }
+        }
+
+        [TestMethod]
+        public void Connecting_To_Blacklisted_Address()
+        {
+            var swarm = new SwarmService {LocalPeer = _self};
+            swarm.BlackList.Add(_mars);
+            swarm.StartAsync().Wait();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() =>
+                {
+                    var _ = swarm.ConnectAsync(_mars).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)